netty/src/main/java/io/grpc/transport/netty/NettyServerTransport.java

```diff
@@ -107,6 +109,11 @@ public void operationComplete(ChannelFuture future) throws Exception {
     channel.pipeline().addLast(streamRemovalPolicy);
     channel.pipeline().addLast(handler);
 
+    GrpcSession session = new GrpcSession(channel.remoteAddress());
+    Attribute<GrpcSession> attr = channel.attr(Utils.ATTRIBUTE_KEY_SESSION);
```

Comment:
Instead of using an attribute, adding a new field to NettyServerHandler and passing the GrpcSession (or the remote address) to the constructor is probably simpler and more explicit.